### PR TITLE
CAL-296 Refactor the Jenkinsfile to match updated Jenkins DSL

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Removed scripted parallel stage syntax and TODOs related to the DSL.

Also fixed a typo, removed skip findbugs from commands, updated sonar url and coverage filter.

#### Who is reviewing it? 
@oconnormi @mackncheesiest @vinamartin @blen-desta @LinkMJB 

#### Select relevant component teams: 
@codice/build 

#### Choose 2 committers to review/merge the PR. 
@clockard @shaundmorris

#### How should this be tested? (List steps with links to updated documentation)
These changes were run in the local test jenkins under "matt test multibranch".

#### What are the relevant tickets?
[CAL-296](https://codice.atlassian.net/browse/CAL-296)

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -180,78 +163,70 @@ pipeline {
...
+                    }
+                }
+                stage ('Codecov') {
+                    agent { label 'linux-small' }
+                    steps {
+                        retry(3) {
+                            checkout scm
+                        }
+                        withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
+                            withCredentials([string(credentialsId: 'DDF_CodeCov', variable: 'CODECOV_TOKEN')]) {
```
Comment (LinkMJB) [inline Jenkinsfile]: @mcalcote I realize this was in use prior as well, but are these the correct credentials for Alliance?

File: Jenkinsfile [hunk 2]
```diff
@@ -48,129 +49,112 @@ pipeline {
...
+                    agent { label 'server-2016-large' }
+                    steps {
+                        retry(3) {
+                            checkout scm
+                        }
+                        timeout(time: 1, unit: 'HOURS') {
+                            withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS}', options: [artifactsPublisher(disabled: true), dependenciesFingerprintPublisher(disabled: true, includeScopeCompile: false, includeScopeProvided: false, includeScopeRuntime: false, includeSnapshotVersions: false)]) {
+                                bat 'mvn install -B -pl !%DOCS% -DskipStatic=true -DskipTests=true $DISABLE_DOWNLOAD_PROGRESS_OPTS -T 1C'
+                                bat 'mvn clean install -B -T 1C -pl !%ITESTS% -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/%CHANGE_TARGET% $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+                                bat 'mvn install -B -pl %ITESTS% -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS'
```
Comment (mackncheesiest) [inline Jenkinsfile]: Does this need to be `%DISABLE_DOWNLOAD_PROGRESS_OPTS%` to account for being windows?

File: Jenkinsfile (hunk 1 above)
Comment (mcalcote) [reply Jenkinsfile]: @shaundmorris helped me update this with Alliance specific credentials

File: Jenkinsfile (hunk 2 above)
Comment (mcalcote) [reply Jenkinsfile]: yes it does ;)

**Comment by shaundmorris** [comment]
CI build not needed for this pr since the Jenkins file isn't used in pr builds right now.

**Pull request merged**